interview-questions.md

```diff
@@ -0,0 +1,52 @@
+# ASSESSMENT 4: Interview Practice Questions
+
+Answer the following questions.
+
+First, without external resources. Challenge yourself to answer from memory.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. What is object-oriented programming? How is it different than functional programming?
+
+Your answer: From my understanding, that means that most(if not all things are objects) I think this would be slightly easier to mannage and just apply classes to the objects instead of having to create the objects as well.
+
+Researched answer: OOP was made to make it easier to work with large software systems. a way to create containers for data that can be changed and manipulated without messing up the rest of the program.
+
+2. What is the difference between a Float and an Integer in Ruby?
+
+Your answer: an integer is a whole number. and float is a number with a decimal point.
+
+Researched answer: integer is a number without a decimal point. A float is a floating point number and has a decimal place. It seems the difference is that ruby assumes an integer is being used and in order to get a float, you need to include a float in the process. i.e. 1 + 1.5
+
+3. Ruby has an implicit return. What does that mean?
+
+Your answer: if there is no return created, the last command will be automatically returned. like a default.
+
+Researched answer: if there is no return "explicitly" prompted within a method, the value of the last executed instruction is returned. this is a good way to keep track of changes in code or to remind the developer to include a return command.
+
+4. What is a block in Ruby?
+
+Your answer: like an anonymous function.
```

Comment:
I would really recommend expanding on these initial answers.  Short answers like this won't show that you are interested or engaged with the interviewer.